### ISSUE TITLE
Configuration of Mautic in Apache subfolder - File Manager not working

### ISSUE THREAD

**Issue by mdelanoe** [mautic/mautic#1381]
Hi!

I am trying to set up Mautic on Ubuntu server 14.04, with Apache 2.4.7 and PHP 5.5.9.
I would like to have Mautic running in a subfolder of my main website:
- My website url: http://www.mywebsite.com/
- Mautic url: http://www.mywebsite.com/pages/

My Apache configuration is as follow:
```
<VirtualHost *:80>
      ServerName www.mywebsite.com

      DocumentRoot /path/to/main/website
      <Directory /path/to/main/website>
         # Main website conf
      </Directory>

      Alias /pages /path/to/mautic
      <Directory /path/to/mautic>
         Require all granted
         AllowOverride all
         Options +FollowSymLinks -MultiViews
      </Directory>
</VirtualHost>
```

My Mautic configuration:
```
Site url: http://www.mywebsite.com/pages
Relative path to the images directory: media/images
```

Everything works well, except for the file manager when creating Landing pages or Emails.

The problem is that filemanager relies on the document root of the server, which in my case doesn't help at all since my Mautic installation is physically not in a subfolder of my main website. Filemanager is trying to open the "/path/to/main/website/pages/media/images" directory which of course doesn't exists (instead of "/path/to/mautic/media/images").

Is there any known way to configure Mautic to work in an Apache Alias, not physically stored in a subdirectory of the document root of the server?
Or is it not possible due to how the filemanager library works?

(I tried to modify Mautic to use the kernel_root_dir instead of the document root of the server when initializing fileManager, which worked and allowed me to browse the files, but I ended up with new issues related to how fileManager generates the assets urls; those would have been resolved if it was possible to set fileManager's baseUrl via PHP, but it doesn't seem to be: see https://github.com/simogeo/Filemanager/issues/339#issuecomment-113034977 where Alan commented)

My only "working solution" so far, is to add a symlink in my main website directory, pointing to my mautic directory. But not ideal, since I would like to keep my 2 websites separated as much as possible.

Thanks in advance

**Comment by escopecz** [comment]
I'm maybe missing a point why you want to do it as a subfolder when it is not a subfolder. You can do the same with a subdomain and have it completely separated from your website.

My explanation about why it doesn't work is that PHP possibly doesn't know about the Apache configuration you've made. I'm not sure this is something we could solve from Mautic side.

**Comment by mdelanoe** [comment]
Thanks @escopecz for your reply.

The reason I want Mautic to be accessible through a subfolder of my main website is to improve SEO for my landing pages (my landing pages would benefit from my main website "score" and vice-versa). So that's why I don't want to set it up on a subdomain (which I tried and it perfectly worked).

But I still wanted to "technically" separate the two websites, for flexibility, maintenance and security reasons. 

PHP seems to read the DocumentRoot Apache directive which is then used by Mautic to configure fileManager.

I'm not necessarily looking for a fix in Mautic itself, but maybe an Apache/PHP configuration trick I missed to bypass that DocumentRoot issue. Maybe someone encountered that use-case before and successfully made it work?

**Comment by npracht** [comment]
Hi, similar issue for me.

## Description
I meet the same issue when i wanna create a subfolder in the image manager. And add pictures or navigate in it. Check picture hereunder.

![capture d ecran 2016-04-19 a 12 26 06](https://cloud.githubusercontent.com/assets/14075239/14636168/3c9d0c70-062a-11e6-8e02-fb6407e1d76c.png)


## Steps to reproduce

1. Open the image manager in your
1.  email WYSIWYG editor,
2. Create a folder,
3. Click on it to get it
4. You'll get the message "No Way"

**Comment by godza** [comment]
Have the same issue.

**Comment by escopecz** [comment]
Guys, could you test it in Mautic 2.1.0? There has been some fixes done in this direction.

**Comment by alanhartless** [comment]
This should be fixed in 2.1.1 with https://github.com/mautic/mautic/pull/2341. Let us know otherwise. Thanks!

**Comment by jeanasturgill** [comment]
Forms not working? 

My issue was either the forms stuck on please wait if I set the site to the subfolder in the configuration OR they worked and the links in the emails and text messages did not work if I used only the domain. 

If you've installed it in a subdirectory / subfolder, leave it there, AND create a subdomain. Be sure to point the subdomain to the folder in the subdirectory. Change the configuration to be the subdomain. Be sure to clear the cache, rebuild the forms, and update the links on any webpages for the forms to the subdomain. You may also need to clear your browser to see any change. 

It may be necessary to use the subdomain as your login when generating forms.
